ublox_serialization/include/ublox_serialization/serialization.hpp

```diff
@@ -532,9 +538,17 @@ class Reader {
     return (classId() == class_id && messageId() == message_id);
   }
 
- private:
+  const std::string &getExtraData() const {
+    const std::string buffer = extra_data_;
+    extra_data_.clear()
```

Comment:
Actually, this doesn't work because this is a `const` method.

But you are correct; the `Reader` instance only exists for the duration of the callback, so we won't be leaking memory.  In that case, let's just remove this line again.  Once that is done, I'm happy to approve and merge.